java/code/src/com/suse/manager/webui/controllers/admin/EnableSCCDataForwardingController.java

```diff
@@ -0,0 +1,90 @@
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.controllers.admin;
+
+import static com.suse.manager.webui.utils.SparkApplicationHelper.withUser;
+import static spark.Spark.get;
+
+import com.redhat.rhn.common.conf.ConfigDefaults;
+import com.redhat.rhn.common.localization.LocalizationService;
+import com.redhat.rhn.common.security.PermissionException;
+import com.redhat.rhn.common.validator.ValidatorError;
+import com.redhat.rhn.domain.role.RoleFactory;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.frontend.action.satellite.SCCDataForwardingConfigAction;
+import com.redhat.rhn.manager.satellite.ConfigureSatelliteCommand;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import spark.ModelAndView;
+import spark.Request;
+import spark.Response;
+import spark.template.jade.JadeTemplateEngine;
+
+/**
+ * Controller class providing backend code to enable SCC data forwarding.
+ */
+public class EnableSCCDataForwardingController {
+
+    private EnableSCCDataForwardingController() { }
+
+    private static final SCCDataForwardingConfigAction CONFIG_ACTION = new SCCDataForwardingConfigAction();
+
+    /**
+     * Initializes the routes.
+     *
+     * @param jade the jade engine
+     */
+    public static void initRoutes(JadeTemplateEngine jade) {
+        get("/manager/admin/enable-scc-data-forwarding", withUser(EnableSCCDataForwardingController::enable), jade);
+    }
+
+    /**
+     * Route to enable data forwarding
+     * @param request - the request
+     * @param response - the response
+     * @param user - the user
+     * @return - jade template with confirmation message
+     */
+    public static ModelAndView enable(Request request, Response response, User user) {
+        String message = LocalizationService.getInstance().getMessage("scc-data-forwarding.enabled");
+
+        if (!user.hasRole(RoleFactory.ORG_ADMIN)) {
+            throw new PermissionException("Only Org Admins can change SCC data forwarding property.");
+        }
+
+        if (ConfigDefaults.get().isForwardRegistrationEnabled()) {
+            message = LocalizationService.getInstance().getMessage("scc-data-forwarding.not-changed");
+        }
+        else {
+            ConfigureSatelliteCommand csc = CONFIG_ACTION.getCommand(user);
+            csc.updateBoolean(ConfigDefaults.FORWARD_REGISTRATION, Boolean.TRUE);
+            ValidatorError[] verrors = csc.storeConfiguration();
```

Comment:
Does the change also reflected in the file(rhn.conf? )?

Reply:
One more question. We are only changing the property here but if some user actually disabled the taskomatic job, we don't take care of enabling that here, right?

Reply:
About the taskomatic job: I think one thing that we could consider doing here is resetting the schedule to the default one (the one you have after installation), just in case someone might have changed it to only run once per month or so? Not a blocker for this PR though, if it turns out to be complex we can skip it for now.

Reply:
If I remember correctly we have a different problem. You cannot "disable" a job. In fact it is removed and you cannot get it back. I think we have a separate issue or bug report about it. This problem is bigger than just a changed schedule

Reply:
https://github.com/SUSE/spacewalk/issues/16996

Reply:
thank you for the pointer Michael, that one is concerning :(